src/diff/children.js

```diff
@@ -56,162 +58,196 @@ export function diffChildren(
 		}
 	}
 
-	i = 0;
-	newParentVNode._children = toChildArray(
-		newParentVNode._children,
-		childVNode => {
-			if (childVNode != null) {
-				childVNode._parent = newParentVNode;
-				childVNode._depth = newParentVNode._depth + 1;
+	newParentVNode._children = [];
+	for (i = 0; i < renderResult.length; i++) {
+		childVNode = renderResult[i];
 
-				// Check if we find a corresponding element in oldChildren.
-				// If found, delete the array item by setting to `undefined`.
-				// We use `undefined`, as `null` is reserved for empty placeholders
-				// (holes).
-				oldVNode = oldChildren[i];
+		if (childVNode == null || typeof childVNode == 'boolean') {
```

Comment:
This big if statement is the internal VNode coercion that `toChildArray` did. Terser converts into a massive ternary for us :)